Hymns/AppDelegate.swift

```diff
@@ -22,6 +23,28 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Called when the user discards a scene session.
         // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
         // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
+
+        //Handle Realm Migrations https://realm.io/docs/swift/latest/#migrations
+        let config = Realm.Configuration(
+            // Set the new schema version. This must be greater than the previously used
+            // version (if you've never set a schema version before, the version is 0).
+            schemaVersion: 1,
+
+            // Set the block which will be called automatically when opening a Realm with
+            // a schema version lower than the one set above
+            migrationBlock: { migration, oldSchemaVersion in
+                // We haven’t migrated anything yet, so oldSchemaVersion == 0
+                if oldSchemaVersion < 1 {
+                    migration.enumerateObjects(ofType: TagEntity.className()) { _, newObject in
+                        newObject!["privateTagColor"] = 0
+                    }
+                    // Nothing to do!
+                    // Realm will automatically detect new properties and removed properties
+                    // And will update the schema on disk automatically
+                }
+        })
+        // Tell Realm to use this new configuration object for the default Realm
+        Realm.Configuration.defaultConfiguration = config
```

Comment:
Hmmm, we have multiple Realm files in our app. Do you know how this would work? This looks like it's making this config the default config across the board for all the realm files in our app, so wouldn't this also affect the history file as well?
https://stackoverflow.com/a/54249204

Reply:
I mean, I think it would be the same as now actually in a sense. Because right now they are still all being rooted from the same Realm.Configuration.defaultConfiguration anyways. We are just calling the .fileUrl method on the existing default configuration. 

But yeah, to answer your question it would affect all of our realm files. Which may be a good thing because there will just be one place for all migration versions. So it's not like we have them staged at various versions. Idk.

Reply:
> But yeah, to answer your question it would affect all of our realm files. Which may be a good thing because there will just be one place for all migration versions.
I think having all the migration code in one place is good, but having a schema upgrade for one realm file affect all the realm files, I don't think is a good thing.
 
My concern is like, say we change Tags to version 2, now do we need to provide a schema version for the history store as well? It's confusing because we are using one schema version to keep track of potentially multiple realm files.

So, let's say we increment the schema version from 1 to 2. Now both the History and Tags will run the migration code. Best case here, it just waste CPU cycles, worse case, there are funky race conditions because they are both trying to increment the TagEntity. I suspect the former will happen because they are different realm files.

I'd be more comfortable if you created a configuration per realm file, and just had the migration code specific to that realm file. In this case, you can just create the configuration file for Tags, since we haven't had the need to do any migrations for History or Favorites.

To your point about having all migration code in the same place, it's a very good point. We can create a `Migrations` class in `Infrastructure/Data` and put all the migration code there. That way we can unit test also!

```
let config = Realm.Configuration(
    // Set the new schema version. This must be greater than the previously used
    // version (if you've never set a schema version before, the version is 0).
    schemaVersion: 1,

    // Set the block which will be called automatically when opening a Realm with
    // a schema version lower than the one set above
    migrationBlock: { migration, oldSchemaVersion in
        // We haven’t migrated anything yet, so oldSchemaVersion == 0
        if oldSchemaVersion < 1 {
            Migrations.performTagsMigration()
        }
})

let realm = try! Realm(fileURL: url!, configuration: config)
return TagStoreRealmImpl(realm: realm) as TagStore
```

https://exceptionshub.com/realm-migrations-in-swift.html

Reply:
I made some changes. Please check my work though. Not sure if I am doing this right.

Reply:
lol bro. I realized that we haven't launched Tags to anybody yet... So we literally don't even need a migration plan. Migrations are for people who have an older version and upgrade, but nobody has the older version >.>

I just added some stub migration configs for each of the tag stores, but we don't need any real migration code at all lol. All we need to do is uninstall the app from our emulator when the version has a mismatch